File: LNPG_01_TIPOS_E_VARIAVEIS/Java/WeightedAverage.java
package LNPG_01_TIPOS_E_VARIAVEIS.Java;

import java.util.Locale;
import java.util.Scanner;

public class WeightedAverage {
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            double result = 0;
            final int MARKS_NUMBER = 3; 
            double mark;
            double weight;
            double weightSum = 0;
            for (int i = 1; i <= MARKS_NUMBER; i++) {
                System.out.printf("\n%d° nota: ", i);
                mark = scanner.nextDouble();
                System.out.printf("%d° peso: ", i);
                weight = scanner.nextDouble();
                result += (mark * weight);
                weightSum += weight;
            }
            System.out.printf(Locale.US, "\nA média das notas foi: %.1f", result/weightSum);
        } catch (Exception e) {
            System.out.println(e);
        }

    }
}

File: LNPG_01_TIPOS_E_VARIAVEIS/C/WeightedAverage.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    /* code */
    double result;
    const int STUDENT_NUMBER = 3;
    double mark;
    double weight;
    double weightSum;
    for (int index = 0; index < STUDENT_NUMBER; index++) {
        printf("\nNota %d: ", index);
        scanf("%lf", &mark);
        printf("Peso %d: ", index);
        scanf("%lf", &weight);
        result += (mark * weight);
        weightSum += weight;
    };
    printf("A media ponderada foi: %.2f", result / weightSum);
    return 0;
}
